src/shapes.c

```diff
@@ -619,31 +619,60 @@ void DrawRectanglePro(Rectangle rec, Vector2 origin, float rotation, Color color
 {
     if (rlCheckBufferLimit(4)) rlglDraw();
 
-    rlEnableTexture(GetShapesTexture().id);
+    Vector2 bl = { 0 };
+    Vector2 br = { 0 };
+    Vector2 tr = { 0 };
+    Vector2 tl = { 0 };
 
-    rlPushMatrix();
-        rlTranslatef(rec.x, rec.y, 0.0f);
-        rlRotatef(rotation, 0.0f, 0.0f, 1.0f);
-        rlTranslatef(-origin.x, -origin.y, 0.0f);
+    // Only calculate rotation if needed
+    if (rotation == 0.0f)
```

Comment:
Be careful, this check could result in false positives due to float resolution issues.

Reply:
Not sure how to solve it... I did a bit of research and it seems that the specific case for 0.0f should work ok... but no clear answer on this topic.

Reply:
It should work fine for functions that pass the rotation as 0.0f internally such as DrawTexture though I am not sure about situations where the user passes in rotation that is very close to 0.0f.